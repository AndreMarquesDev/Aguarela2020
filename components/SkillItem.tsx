import React, { FC } from 'react';
import Image from 'next/image';

interface SkillItemProps {
    icon: string;
    title: string;
    description: string;
}

const SkillItem: FC<SkillItemProps> = ({ icon, title, description }) => (
    <>
        <Image
            alt={title}
            className="imageOpacity"
            height={130}
            layout="fixed"
            src={`/images/icons/${icon}.svg`}
            width={160}
        />
        <strong>{title}</strong>
        <p>{description}</p>

        <style jsx>
            {`
                @import './styles/_vars.scss';

                strong {
                    @include fontM($fontWeight: 500);
                    letter-spacing: 5rem;
                    margin: 40rem 0 15rem;
                }

                p {
                    @include fontXS;
                    letter-spacing: 1rem;
                    line-height: 19rem;
                }
            `}
        </style>
    </>
);

export default SkillItem;
