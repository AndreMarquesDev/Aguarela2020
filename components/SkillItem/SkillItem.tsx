import React from 'react';
import Image from 'next/image';

export interface SkillItemProps {
    icon: string;
    title: string;
    description: string;
}

export const SkillItem = ({ icon, title, description }: SkillItemProps): JSX.Element => (
    <>
        <Image
            alt={title}
            className="imageOpacity"
            height={130}
            layout="fixed"
            src={`/images/icons/${icon}.svg`}
            width={160}
        />
        <strong>{title}</strong>
        <p>{description}</p>

        <style jsx>
            {`
                @import './styles/_vars.scss';

                strong {
                    @include fontM($fontWeight: 500);
                    letter-spacing: 5rem;
                    margin: 40rem 0 15rem;
                }

                p {
                    @include fontXS;
                    letter-spacing: 1rem;
                    line-height: 19rem;
                }
            `}
        </style>
    </>
);
