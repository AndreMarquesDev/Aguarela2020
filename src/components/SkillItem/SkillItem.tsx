import React from 'react';
import Image from 'next/image';

export interface SkillItemProps {
    icon: string;
    title: string;
    description: string;
}

export const SkillItem = ({ icon, title, description }: SkillItemProps): JSX.Element => (
    <>
        <Image
            alt={title}
            className="imageOpacity"
            height={130}
            src={`/images/icons/${icon}.svg`}
            width={160}
        />
        <strong className="fontM">{title}</strong>
        <p className="fontXS">{description}</p>

        <style jsx>
            {`
                strong {
                    font-weight: 500;
                    letter-spacing: 5rem;
                    margin: 40rem 0 15rem;
                }

                p {
                    letter-spacing: 1rem;
                    line-height: 19rem;
                }
            `}
        </style>
    </>
);
