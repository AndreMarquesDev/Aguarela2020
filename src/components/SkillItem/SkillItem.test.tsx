import type { RenderResult } from '@testing-library/react';
import type { SkillItemProps } from './SkillItem';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { SkillItem } from './SkillItem';
import '@testing-library/jest-dom';

const baseProps: SkillItemProps = {
    icon: 'strategy',
    title: 'Título',
    description: 'Descrição',
};

const renderComponent = (newProps?: Partial<SkillItemProps>): RenderResult => {
    return render(<SkillItem {...baseProps} {...newProps} />);
};

describe('<SkillItem />', () => {
    it('renders properly', () => {
        const { container } = renderComponent();

        expect(screen.getByText(baseProps.title)).toBeInTheDocument();
        expect(screen.getByText(baseProps.description)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
