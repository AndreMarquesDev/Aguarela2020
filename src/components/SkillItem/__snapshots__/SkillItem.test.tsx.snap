// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkillItem /> renders properly 1`] = `
<div>
  <span
    style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 160px; height: 130px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
  >
    <img
      alt="Título"
      class="imageOpacity"
      data-nimg="fixed"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
    <noscript />
  </span>
  <strong
    class="jsx-390248359"
  >
    Título
  </strong>
  <p
    class="jsx-390248359"
  >
    Descrição
  </p>
</div>
`;
